tests/baselines/reference/typeGuardsWithInstanceOfByConstructorSignature.js

```diff
@@ -0,0 +1,137 @@
+//// [typeGuardsWithInstanceOfByConstructorSignature.ts]
+interface AConstructor {
+    new (): A;
+}
+interface A {
+    foo: string;
+}
+declare var A: AConstructor;
+
+var obj1: A | string;
+if (obj1 instanceof A) { // narrowed to A.
+    obj1.foo;
+    obj1.bar;
+}
+
+var obj2: any;
+if (obj2 instanceof A) { // can't type narrowing from any.
+    obj2.foo;
+    obj2.bar;
+}
+
+// with generics
+interface BConstructor {
```

Comment:
A few more tests:
- A construct signature that returns a union type
- a construct signature that returns any
- a type with a prototype that has any type